src/lib/mina_ledger/root.ml

```diff
@@ -85,50 +117,143 @@ struct
+    let () = Unix.mkdir primary_directory in
+    let () = Unix.mkdir converting_directory in
+    Converting_ledger.Config.{ primary_directory; converting_directory }
+
+  let create ~logger ~config:Config.{ top_directory; backing_type } ~depth () =
+    match backing_type with
+    | Stable_db ->
+        Stable_db (Stable_db.create ~directory_name:top_directory ~depth ())
+    | Converting_db ->
+        let config = prepare_converting_dirs top_directory in
+        Converting_db
+          (Converting_ledger.create ~config:(In_directories config) ~logger
+             ~depth () )
+
+  let create_temporary ~logger ~backing_type ~depth () =
+    match backing_type with
+    | Config.Stable_db ->
+        Stable_db (Stable_db.create ~depth ())
+    | Converting_db ->
+        Converting_db
+          (Converting_ledger.create ~config:Temporary ~logger ~depth ())
+
+  let create_checkpoint t ~config:Config.{ top_directory; backing_type } () =
+    match t with
+    | Stable_db db ->
+        if not Config.(equal_backing_type backing_type Stable_db) then
+          raise
+            (Config.Backing_mismatch
+               { backing_1 = Stable_db; backing_2 = backing_type } )
+        else
+          Stable_db
+            (Stable_db.create_checkpoint db ~directory_name:top_directory ())
+    | Converting_db db ->
+        if not Config.(equal_backing_type backing_type Converting_db) then
+          raise
+            (Config.Backing_mismatch
+               { backing_1 = Converting_db; backing_2 = backing_type } )
+        else
+          let config = prepare_converting_dirs top_directory in
+          Converting_db (Converting_ledger.create_checkpoint db ~config ())
+
+  let make_checkpoint t ~config:Config.{ top_directory; backing_type } =
+    match t with
+    | Stable_db db ->
+        if not Config.(equal_backing_type backing_type Stable_db) then
+          raise
+            (Config.Backing_mismatch
+               { backing_1 = Stable_db; backing_2 = backing_type } )
+        else Stable_db.make_checkpoint db ~directory_name:top_directory
+    | Converting_db db ->
+        if not Config.(equal_backing_type backing_type Converting_db) then
+          raise
+            (Config.Backing_mismatch
+               { backing_1 = Converting_db; backing_2 = backing_type } )
+        else
+          let config = prepare_converting_dirs top_directory in
+          Converting_ledger.make_checkpoint db ~config
 
   let as_unmasked t =
-    match t with Stable_db db -> Any_ledger.cast (module Stable_db) db
+    match t with
+    | Stable_db db ->
+        Any_ledger.cast (module Stable_db) db
+    | Converting_db db ->
+        Any_ledger.cast (module Converting_ledger) db
 
   let transfer_accounts_with ~stable ~src ~dest =
     match (src, dest) with
     | Stable_db db1, Stable_db db2 ->
         stable ~src:db1 ~dest:db2 |> Or_error.map ~f:(fun x -> Stable_db x)
+    | _ ->
+        failwith "TODO: this function should be removed"
```

Comment:
I see in https://github.com/MinaProtocol/mina/pull/17659, this is removed. Hence didn't take time to implement it.